### PR TITLE
Fix "can only test a child process"

### PR DESCRIPTION
**Pull request by farizrahman4u**
Description
-----------

Allow logging from forked process:

```python
import wandb
import time
import multiprocessing as mp



def train():
    time.sleep(1)
    wandb.log({"acc": 0.9})


if __name__ == "__main__":
    wandb.init()
    nproc = 2
    procs = []
    for i in range(nproc):
        procs.append(mp.Process(target=train, daemon=True))
    for p in procs:
        p.start()
    for p in procs:
        p.join()

```

Testing
-------

How was this PR tested? manually

### CODE REVIEW

File: wandb/sdk/interface/interface.py [hunk 1]
```diff
@@ -513,8 +514,16 @@ def _make_record(
             raise Exception("Invalid record")
         return record
 
+    def _is_process_alive(self) -> bool:
+        if not self._process:
+            return True
```
Comment (vanpelt) [inline wandb/sdk/interface/interface.py:519]: Do we want to assume the process is alive if not `self._process`?

Comment (farizrahman4u) [reply wandb/sdk/interface/interface.py:519]: That is the current logic:

```python
        if self._process and not self._process.is_alive():
            raise Exception("The wandb backend process has shutdown")
```

Also stuff crashes without that assumption, I checked. (Granted we are actually checking `is_processes_not_dead()` :p)

Comment (vanpelt) [reply wandb/sdk/interface/interface.py:519]: Ah, I see.  I guess I'm curious what the scenario is when this is called and there's no process...

**Review by raubitsj — CHANGES_REQUESTED** [review]
I get what this PR is doing and I think it will make things not crash immediately on a particular assertion but it will do the wrong thing in all but a few cases.

Basically the wandb library does not yet support multiprocessing, this is a a bandaid to make it work (sometimes)

We still have locally cached data in the wandb.run object that will be different after the fork.

The intention is to make wandb multiprocessing safe in the future: https://wandb.atlassian.net/browse/WB-4316, but it is not now.

It seems like a better solution is to detect this use case and spit out an error with a link to an application note on our doc site suggesting our 2 multiprocessing workarounds:
1) run wandb.init(). in each process
2) only run wandb.init() in your rank0 process and funnel logs to that process using communication channels appropriate in your environment.

What do you think @vanpelt

**Comment by raubitsj** [comment]
Closing in favor of:
https://github.com/wandb/client/pull/1986

wandb.log() is not safe to call when forked from the wandb.init() process... we should probably ignore all method calls for now, but the above PR starts with wandb.log()

**Pull request closed without merging**